tests/bdd/volgroup_vgpattern_bdd.feature

```diff
@@ -0,0 +1,131 @@
+
+  
+      Scenario: test sc with vgpattern matching no vg
+          Given sc is created with vgpattern "lvmvgn.*"
+          When a pvc is created by referencing the sc
+          Then pvc should not be Bound
+          And lvmvolume should not be created
+
+  
+      Scenario: test pvc with higher size then vg free space
+          Given "lvmvg" has lesser free space then "lvmvg1"
+          And sc is created with volgroup as "lvmvg"
+          When a pvc is created with size higher then "lvmvg" and lower then "lvmvg1" free space
+          Then pvc should not be boun
+          And lvmvolume cr should not be created
+
+  
+  Background:
+      Given a 3 lvmnode cluster is configured
+      And each node <node> has <vg> created with same size 
+
+      Scenario: test sc with volgroup present only on one node
+          Given a sc is created with volgroup as "lvmvg1" with no topology
+          When pvc is created referencing the same sc
+          Then pvc should be in Pending state
+          And lvmvolume cr should not be created
+
+      Scenario: test sc with volgroup present on one node with allowed topology
+          Given a sc is created with volgroup as "lvmvg1"
+          And sc has topology key with "node1" value
+          When pvc is created referencing the same sc
+          Then pvc should be bound 
+          And lvmvolume cr should be Ready
+          And lvmvolume node should be allowed topology node
+
+       Scenario: test sc with volgroup not present on any node
+          Given a sc is created with volgroup as "lvmvgn"
+          When pvc is created referencing the same sc
+          Then pvc should be in Pending state
+          And lvmvolume should not be created
+
+      Examples:
+        |  node        | vg           |
+        | node1        | lvmvg1       |
+        | node2        | lvmvg2       |
+        | node3        | lvmvg3       |
+       
+
+  Background:
+      Given a 3 lvmnode cluster is configured
+      And each node <node> has <vg> created with <size>
+
+      Scenario: test sc with volgroup present on 2 nodeswith varying free space and allowed topology
+          Given an sc is created with volgroup "lvmvg"
+          And node1 and node2 is in allowed topology
+          When a pvc is created with "1.3G" size
+          Then pvc should be Bound
+          And lvmvolume should in Ready state
+          And volume owner should be node2
+
+      Examples:
+        |  node        | vg          |  size    |
+        | node1        | lvmvg       |  1G      |
+        | node2        | lvmvg       |  1.5G    |
+        | node3        | lvmvgdiff   |  1G      |
+
+        
+      Scenario: test sc with volgroup present on 2 nodes with less space and allowed topology
+          Given an sc is created with volgroup "lvmvg"
+          And node1 and node2 is in allowed topology
+          When a pvc is created with "1.3G" size
+          Then pvc should be Pending
+          And lvmvolume should not be created
```

Comment:
How about using a pattern here and make the space on some vg more than needed to see if we select the vg which has enough space?

Reply:
We could, However, we have cases here where there is lots of vgs with carying size. Creating so many PVs with allowed topology just to build the setup seemed cumbersome. Creating a device of specific size andPV and VG on that seemed easier.

Reply:
We can extend one of these scenario to say: _**when the vg is expanded to 1.5G(or simply say more than the required pvc size), the pvc should become bound**_

Reply:
Will do